### PR TITLE
out_es: add path option.

### LINKED ISSUE #900: out_es: connect to an elasticsearch on sub-path
This feature request (and the following PRs) arose from our usecase of Fluent-bit running on each k8s node and delivering logs into a dedicated elastcisearch instance, one ES instance per k8s cluster. In order to minimize operational burden, all ES isntances are hosted behind a reverse proxy on a single domain (and single TLS cert), but each ES has its own sub-path. Fluent-bit have no such possibility to send log to an ES not on root, so previously we used fluentd with "path" option:
https://docs.fluentd.org/v1.0/articles/out_elasticsearch#path-(optional)

Luckily, it proved trivial to add such option into Fluent-bit too. It would be great to have this feature accepted upstream. The PRs will follow.

An alternative solution might be to have one more reverse proxy sitting in the same kubernetes pod together with FLB to rewrite the query path. It is surely doable, but not as elegant as the `Path` option of `out_es`.

### PR DESCRIPTION
**Pull request by kabakaev**
Fixes #900.

### CODE REVIEW

File: plugins/out_es/es_conf.c [hunk 1]
```diff
@@ -227,13 +228,20 @@ struct flb_elasticsearch *flb_es_conf_create(struct flb_output_instance *ins,
         }
     }
 
+    /* Elasticsearch: Path */
+    path = flb_output_get_property("path", ins);
+    if (!path)
```
Comment (edsiper) [inline plugins/out_es/es_conf.c]: Minor style change is required, it should be:

```
    if (!path) {
```

Comment (kabakaev) [reply plugins/out_es/es_conf.c]: Changed. PTAL

**Comment by kabakaev** [comment]
@edsiper , your suggestions are implemented. Build is green. Would you mind to take a look again?

**Comment by edsiper** [comment]
please make sure to release path memory after use it

**Comment by kabakaev** [comment]
> please make sure to release path memory after use it

@edsiper, thanks for catching this.  `path` [is freed](https://github.com/fluent/fluent-bit/pull/901/files#diff-a675ccb59bd1753ac1682f2e8db013bfR254).

**Pull request merged**